x-pack/platform/plugins/shared/alerting/server/rules_client/lib/denormalize_artifacts.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { SavedObjectReference } from '@kbn/core/server';
+import type { Artifact } from '../../types';
+import type { DenormalizedArtifacts } from '../types';
+
+export function denormalizeArtifacts(ruleArtifacts: Artifact | undefined): {
```

Comment:
Could you add tests for this file and also if there are already any tests for extract_references and get_alert_from_raw within this lib directory could you add coverage for this feature?

Reply:
Yep I am working on it!

Reply:
Done in 894deecd.